src/tasks/minions/woodcuttingActivity.ts

```diff
@@ -1,35 +1,262 @@
-import { randInt, Time } from 'e';
-import { Bank } from 'oldschooljs';
+import { percentChance, randInt, Time } from 'e';
+import { Bank, LootTable } from 'oldschooljs';
 
 import { Emoji, Events } from '../../lib/constants';
+import { MediumSeedPackTable } from '../../lib/data/seedPackTables';
 import addSkillingClueToLoot from '../../lib/minions/functions/addSkillingClueToLoot';
+import { eggNest } from '../../lib/simulation/birdsNest';
 import Woodcutting from '../../lib/skilling/skills/woodcutting';
-import { Log, SkillsEnum } from '../../lib/skilling/types';
+import { SkillsEnum } from '../../lib/skilling/types';
 import { WoodcuttingActivityTaskOptions } from '../../lib/types/minions';
 import { perTimeUnitChance, roll, skillingPetDropRate } from '../../lib/util';
 import { handleTripFinish } from '../../lib/util/handleTripFinish';
-import resolveItems from '../../lib/util/resolveItems';
+import { userStatsBankUpdate } from '../../mahoji/mahojiSettings';
 
-function handleForestry({ log, duration, loot }: { log: Log; duration: number; loot: Bank }) {
-	if (resolveItems(['Redwood logs', 'Logs']).includes(log.id)) return;
+interface ForestryEvent {
+	id: number;
+	name: string;
+	uniqueXP: SkillsEnum;
+}
+
+const ForestryEvents: ForestryEvent[] = [
+	{
+		id: 1,
+		name: 'Rising Roots',
+		uniqueXP: SkillsEnum.Woodcutting
+	},
+	{
+		id: 2,
+		name: 'Struggling Sapling',
+		uniqueXP: SkillsEnum.Farming
+	},
+	{
+		id: 3,
+		name: 'Flowering Bush',
+		uniqueXP: SkillsEnum.Woodcutting
+	},
+	{
+		id: 4,
+		name: 'Woodcutting Leprechaun',
+		uniqueXP: SkillsEnum.Woodcutting
+	},
+	{
+		id: 5,
+		name: 'Beehive',
+		uniqueXP: SkillsEnum.Construction
+	},
+	{
+		id: 6,
+		name: 'Friendly Ent',
+		uniqueXP: SkillsEnum.Fletching
+	},
+	{
+		id: 7,
+		name: 'Poachers',
+		uniqueXP: SkillsEnum.Hunter
+	},
+	{
+		id: 8,
+		name: 'Enchantment Ritual',
+		uniqueXP: SkillsEnum.Woodcutting
+	},
+	{
+		id: 9,
+		name: 'Pheasant Control',
+		uniqueXP: SkillsEnum.Thieving
+	}
+];
+
+const leafTable = new LootTable()
+	.add('Leaves', 20)
+	.add('Oak Leaves', 20)
+	.add('Willow Leaves', 20)
+	.add('Maple Leaves', 20)
+	.add('Yew Leaves', 20)
+	.add('Magic Leaves', 20);
```

Comment:
Item names in our code are always written following the osrs convention (only first word capitalized)